src/control/ToolHandler.cpp

```diff
@@ -270,43 +297,57 @@ void ToolHandler::setLineStyle(const LineStyle& style) {
  * 			and therefore should not be applied to a selection
  */
 void ToolHandler::setColor(Color color, bool userSelection) {
-    if ((this->buttonSelectedTool->capabilities & TOOL_CAP_COLOR) == 0) {
+    if (this->activeTool != this->toolbarSelectedTool && (this->activeTool->capabilities & TOOL_CAP_COLOR) == 0) {
```

Comment:
Why not use `toolHandler->hasCapability(TOOL_CAP_COLOR)`?

Reply:
Honestly, I think when I was writing that piece of Code I probably had forgotten about that function :sweat_smile:

Reply:
This is fixed in 38d509654dd5b017f84bf8b17f59b0a907d83638.